tests/test_theming/test_theming.py

```diff
@@ -126,3 +148,41 @@ def test_theme_sidebars(app, status, warning):
     assert '<h3>Related Topics</h3>' not in result
     assert '<h3>This Page</h3>' not in result
     assert '<h3 id="searchlabel">Quick search</h3>' in result
+
+
+@pytest.mark.parametrize(
+    'theme_name',
+    [
+        'alabaster',
+        'agogo',
+        'basic',
+        'bizstyle',
+        'classic',
+        'default',
+        'epub',
+        'haiku',
+        'nature',
+        'nonav',
+        'pyramid',
+        'scrolls',
+        'sphinxdoc',
+        'traditional',
+    ],
```

Comment:
Ok, that's impressive - my initial guess after finding that hardly any of the HTML output from the `sphinx13` theme (Sphinx's own `doc` theme) parsed as XML was that most other themes would not parse.

I have to think about this a bit more - I sense that @picnixz had some doubts/considerations in mind too, but I think may be unavailable for a little while.  I'll catch up on this again in the next few days.

In any case, I think I will close my PR - this is definitely way more thorough.